packages/lesmis/src/views/DashboardReportTabView.js

```diff
@@ -14,18 +14,41 @@ import {
   TabBarSection,
   FlexCenter,
   Report,
-  TabBar,
+  StickyTabBar,
   Tabs,
   Tab,
   TabPanel,
   YearSelector,
 } from '../components';
-import { DEFAULT_DASHBOARD_GROUP, DEFAULT_DATA_YEAR } from '../constants';
+import { DEFAULT_DASHBOARD_GROUP, DEFAULT_DATA_YEAR, NAVBAR_HEIGHT_INT } from '../constants';
 
 const DashboardSection = styled(FlexCenter)`
   min-height: 31rem;
 `;
 
+const ScrollToTopButtonContainer = styled.div`
+  position: sticky;
+  bottom: 29px;
+  padding-right: 32px;
+  cursor: pointer;
+  display: flex;
+  justify-content: flex-end;
+`;
+
+const ScrollToTopButtonImg = styled.img`
+  width: 40px;
+  height: 40px;
+`;
+
+const ScrollToTopButton = React.memo(({ onClick }) => (
+  <ScrollToTopButtonContainer onClick={onClick}>
+    <ScrollToTopButtonImg src="/images/up-arrow.svg" alt="back to top" />
+  </ScrollToTopButtonContainer>
+));
+ScrollToTopButton.propTypes = {
+  onClick: PropTypes.func.isRequired,
+};
+
```

Comment:
```suggestion
const useRefHeight = () => {
  const [tabBarHeight, setTabBarHeight] = useState(0);

  const measureTabBarHeight = useCallback(tabBarNode => {
    if (tabBarNode !== null) {
      setTabBarHeight(tabBarNode.getBoundingClientRect().height);
    }
  }, []);

  return [tabBarHeight, measureTabBarHeight];
};

Even though it makes sense for the behaviour to be in with this component, it might still be nice to pull out the code for getting the height of the toolbar into a hook above just for readability reasons. I've put an example here but it would need to change a bit if the forward reffing was removed.
```

Reply:
Even though it makes sense for the behaviour to be in with this component, it might still be nice to pull out the code for getting the height of the toolbar into a hook above just for readability reasons. I've put an example here but it would need to change a bit if the forward reffing was removed.

Reply:
Agreed - it was getting a bit too complex in that component and this was a nice option for pulling out. Done.